docs/website/content/v0.3/en/guides/cloud/azure.md

```diff
@@ -2,145 +2,277 @@
+              --query [ipAddress] \
+              --output tsv)
 
-# Generate a default Talos config for this cluster
-# # This should generate master-{1,2,3}.yaml, worker.yaml, and talosconfig in your PWD
-./osctl config generate cluster.local $MASTERIPS
+osctl config generate talos-k8s-azure-tutorial https://${LB_PUBLIC_IP}:6443
 ```
 
-## Compute Creation
+### Compute Creation
+
+We are now ready to create our azure nodes.
 
 ```bash
-# Create master nodes
-# # `--admin-username` and `--generate-ssh-keys` are required by the az cli,
-# # but are not actually used by talos
-# # `--os-disk-size-gb` is the backing disk for Kubernetes and any workload containers
-# # `--boot-diagnostics-storage` is to enable console output which may be necessary
-# # for troubleshooting
+# Create availability set
+az vm availability-set create \
+  --name talos-controlplane-av-set \
+  -g $GROUP
+
+# Create controlplane 0
 az vm create \
-  --name talos1 \
+  --name talos-controlplane-0 \
   --image talos \
-  --custom-data ./master-1.yaml \
-  --public-ip-address talos-master-1 \
+  --custom-data ./init.yaml \
   -g $GROUP \
   --admin-username talos \
   --generate-ssh-keys \
   --verbose \
   --boot-diagnostics-storage $STORAGE_ACCOUNT \
-  --nsg talos \
-  --os-disk-size-gb 64 \
+  --os-disk-size-gb 20 \
+  --nics talos-controlplane-nic-0 \
+  --availability-set talos-controlplane-av-set \
   --no-wait
-az vm create --name talos2 --image talos --custom-data ./master-2.yaml  --public-ip-address talos-master-2 -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
-az vm create --name talos3 --image talos --custom-data ./master-3.yaml  --public-ip-address talos-master-3 -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
 
-## Create worker nodes, reuse as needed
-az vm create --name talos4 --image talos --custom-data ./worker.yaml -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
+# Create 2 more controlplane nodes
+for i in $( seq 1 2 ); do
+  az vm create \
+    --name talos-controlplane-$i \
+    --image talos \
+    --custom-data ./controlplane.yaml \
+    -g $GROUP \
+    --admin-username talos \
+    --generate-ssh-keys \
+    --verbose \
+    --boot-diagnostics-storage $STORAGE_ACCOUNT \
+    --os-disk-size-gb 20 \
+    --nics talos-controlplane-nic-$i \
+    --availability-set talos-controlplane-av-set \
+    --no-wait
+done
+
+# Create worker node
+  az vm create \
+    --name talos-worker-0 \
+    --image talos \
+    --custom-data ./join.yaml \
+    -g $GROUP \
+    --admin-username talos \
+    --generate-ssh-keys \
+    --verbose \
+    --boot-diagnostics-storage $STORAGE_ACCOUNT \
+    --nsg talos-sg \
+    --os-disk-size-gb 20 \
+    --no-wait
+
+# NOTES:
```

Comment:
nit: We have avoided documentation things in comments. These should be part of the documentation and not the code block IMO.